ispyb/cli/last_data_collections_on.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import os
+import pathlib
+import sys
+import time
+
+from sqlalchemy.orm import Load
+
+import ispyb
+from ispyb.sqlalchemy import BLSession, DataCollection, GridInfo, Proposal
+
+
+def print_data_collections(rows, link=False):
+    for row in reversed(rows):
+        visit = f"{row.Proposal.proposalCode}{row.Proposal.proposalNumber}-{row.BLSession.visit_number}"
+        bl_name = row.BLSession.beamLineName
+        n_images = row.DataCollection.numberOfImages
+        dcid = row.DataCollection.dataCollectionId
+        template = (
+            pathlib.Path(row.DataCollection.imageDirectory)
+            / row.DataCollection.fileTemplate
+        )
+        start_time = f"{row.DataCollection.startTime:%Y-%m-%d %H:%M}"
+        grid_size = (
+            f"{row.GridInfo.steps_x:.0f}x{row.GridInfo.steps_y:.0f}"
+            if row.GridInfo
+            else ""
+        )
+        grid = f", {grid_size:>5} grid" if grid_size else ""
+        print(
+            f"{start_time} {bl_name:8} {dcid:8} {visit:<11} {n_images:4} images{grid}   {template}"
+        )
+        if link:
+            print(" " * 52 + f"https://ispyb.diamond.ac.uk/dc/visit/{visit}/id/{dcid}")
```

Comment:
Is there a place in the database that determines the base URL? I'm just thinking that hardcoding this isn't very portable.

Reply:
It appears not, however you can now override the base SynchWeb URL via the command line, e.g.:

`ispyb.last_data_collections_on i03 --link --synchweb-url=https://wls.ac.uk`

https://github.com/DiamondLightSource/ispyb-api/pull/123/commits/ac652f51e14819b1922a2c26394eaa4bc72829c2

Reply:
Seen it, thank you. It was actually just literally meant as a question to @KarlLevik on how we handle... localisation (for lack of a better term) in ISPyB. I could see that to be useful in more than just this one context.